modules/transitive/src/closure/TransitiveModel.scala

```diff
@@ -53,25 +74,57 @@ object TransitiveModel {
         .map(_.asScala.toSet)
         .fold[Set[Shape]](Set.empty)(_ ++ _)
 
-    val allShapes = (closure ++ entrypointTraits)
+    val allShapes =
+      (closure ++ entrypointTraits).filter(s => !visitedShapes.contains(s))
+    entryPointsShapes.foreach(visitedShapes.add)
```

Comment:
I was kind of expecting we'd do `allShapes.foreach(visitedShapes.add)`

Reply:
also, probably more efficient to do `visitedShapes ++= thing`

Reply:
addressed this at [b0eb2c8](https://github.com/disneystreaming/smithy-translate/pull/42/commits/b0eb2c862445b86ba4c38c236886e0664cab7d04)